### PR TITLE
TWI-48 Reply

### PR DESCRIPTION
**Pull request by pm888**
Reply tweet with parent id

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -248,7 +248,40 @@ paths:
...
+      security:
+        - apiKey: [ ]
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ReplyRequest'
+      responses:
+        '201':
```
Comment (bunin) [inline openapi.yaml:271]: you do not return 201 code in the code

File: migrations/migrations.go [hunk 2]
```diff
@@ -79,6 +79,17 @@ var items = []darwin.Migration{
 			timestamp TIMESTAMP WITH TIME ZONE NOT NULL
 		)`,
 	},
+	{
+		Version:     7,
+		Description: `reply`,
+		Script: `CREATE TABLE reply (
```
Comment (bunin) [inline migrations/migrations.go]: why not using a main table?

File: internal/tweets/tweetsStr.go [hunk 3]
```diff
@@ -34,3 +34,7 @@ type TweetValid struct {
 	Validate *validator.Validate
 	ValidErr map[string]string
 }
+type TweetReply struct {
+	ReplyId int
```
Comment (bunin) [inline internal/tweets/tweetsStr.go]: this does not seem to be used/needed

**Review by bunin — CHANGES_REQUESTED** [review]
https://github.com/mo-pm888/twitter/pull/37/files#r1259883470

File: internal/tweets/tweetsStr.go (hunk 3 above)
Comment (pm888) [reply internal/tweets/tweetsStr.go]: done

File: internal/services/helpers.go [hunk 4]
```diff
@@ -134,3 +137,25 @@ func ReturnErr(w http.ResponseWriter, err string, code int) {
 	w.WriteHeader(code)
 	json.NewEncoder(w).Encode(errj)
 }
+func CreatNewTweet(newTweet tweets.CreatNewTweet, r *http.Request, userID int, tweetTo string) error {
```
Comment (bunin) [inline internal/services/helpers.go]: passing request around is not a good idea, if you only need a context pass a context

File: internal/services/helpers.go [hunk 5]
```diff
@@ -134,3 +137,25 @@ func ReturnErr(w http.ResponseWriter, err string, code int) {
...
+	query := `INSERT INTO tweets (user_id, text, created_at, public, only_followers, only_mutual_followers, only_me,reply_to)
+		VALUES ($1, $2, $3, $4, $5, $6, $7,&8) RETURNING tweet_id`
```
Comment (bunin) [inline internal/services/helpers.go]: `&8` ???

File: internal/tweets/tweetsStr.go (hunk 3 above)
Comment (bunin) [reply internal/tweets/tweetsStr.go]: done where?

File: internal/users/users.go [hunk 6]
```diff
@@ -10,21 +10,42 @@ import (
...
+	validate *validator.Validate
+	validErr map[string]string
+}
+type NameVal struct {
+	short    bool
+	long     bool
+	realName bool
+}
+
+func (v *UserValid) Error() string {
```
Comment (bunin) [inline internal/users/users.go:35]: this should not be here

File: internal/users/users.go [hunk 7]
```diff
@@ -309,47 +330,6 @@ func UnfollowUser(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 	fmt.Println("Done")
 }
-
-func EditProfile(w http.ResponseWriter, r *http.Request) {
```
Comment (bunin) [inline internal/users/users.go:306]: this should not be here

**Review by bunin — CHANGES_REQUESTED** [review]

File: internal/tweets/tweetsStr.go (hunk 3 above)
Comment (pm888) [reply internal/tweets/tweetsStr.go]: done

File: internal/services/helpers.go (hunk 4 above)
Comment (pm888) [reply internal/services/helpers.go]: done

File: internal/services/helpers.go (hunk 5 above)
Comment (pm888) [reply internal/services/helpers.go]: done

File: internal/services/helpers.go [hunk 8]
```diff
@@ -134,3 +138,25 @@ func ReturnErr(w http.ResponseWriter, err string, code int) {
 	w.WriteHeader(code)
 	json.NewEncoder(w).Encode(errj)
 }
+func CreatNewTweet(newTweet tweets.CreatNewTweet, ctx context.Context, userID int, tweetTo string) error {
```
Comment (bunin) [inline internal/services/helpers.go]: typo

File: internal/server/server.go [hunk 9]
```diff
@@ -30,9 +30,9 @@ func Server() {
 	r.HandleFunc("/v1/users/{id}/unfollow", func(w http.ResponseWriter, r *http.Request) {
 		Serviceuser.AuthHandler(http.HandlerFunc(Serviceuser.UnfollowUser)).ServeHTTP(w, r)
 	}).Methods(http.MethodPost)
-	r.HandleFunc("/v1/users", func(w http.ResponseWriter, r *http.Request) {
-		Serviceuser.AuthHandler(http.HandlerFunc(Serviceuser.EditProfile)).ServeHTTP(w, r)
-	}).Methods(http.MethodPatch)
+	//r.HandleFunc("/v1/users", func(w http.ResponseWriter, r *http.Request) {
+	//	Serviceuser.AuthHandler(http.HandlerFunc(Serviceuser.EditProfile)).ServeHTTP(w, r)
+	//}).Methods(http.MethodPatch)
```
Comment (bunin) [inline internal/server/server.go:35]: why is it here?

File: internal/users/users.go [hunk 10]
```diff
@@ -547,3 +542,49 @@ func DeleteUserSession(token string) error {
 
 	return nil
 }
+func RegisterUsersValidations(userValid *UserValid) error {
```
Comment (bunin) [inline internal/users/users.go:545]: this does not seem to have anything to do with Reply

**Review by bunin — CHANGES_REQUESTED** [review]

**Pull request closed without merging**